src/backend/Donate/Actions/MyDonations/GetDetails/GetMyDonationDetailsService.cs

```diff
@@ -0,0 +1,112 @@
+            GetCurrentUserService getCurrentUserService,
+            RoleService roleService,
+            Hashids hashids,
+            ILogger<GetWarehousesService> logger
+            )
+        {
+            _appDbContext = appDbContext;
+            _currentUserService = currentUserService;
+            _getCurrentUserService = getCurrentUserService;
+            _roleService = roleService;
+            _hashids = hashids;
+            _logger = logger;
+        }
+
+        public async Task<GetMyDonationDetailsResponse> GetMyDonationDetails(string id)
+        {
+            var auth0UserInfo = await _currentUserService.GetUserInfo();
+            var currentUser = await _getCurrentUserService.Execute(auth0UserInfo);
+            await _roleService.ThrowIfNoRole(RoleName.Donor, currentUser.Id);
+            var decodedDonationId = _hashids.DecodeSingleLong(id);
+
+
+            var donationById = _appDbContext.Donations.SingleOrDefaultAsync(c => c.Id == decodedDonationId
+               && c.Status == DbEntityStatus.Active || c.Status == DbEntityStatus.Inactive);
+
+            if (donationById == null)
+            {
+                _logger.LogError("Couldn't find donation in database");
+                throw new ApplicationErrorException("Sorry there seems to be a problem with our service");
+            }
+
+            var listOfDonationProducts = await _appDbContext.DonationProducts
+                .Include(c => c.Product)
+                .Where(c => c.Donation == donationById.Result
+                && c.Status == DbEntityStatus.Active).ToListAsync();
+
+            if (!listOfDonationProducts.Any())
+            {
+                _logger.LogError("Couldn't find any active products in database");
+                throw new ApplicationErrorException("Sorry there seems to be a problem with our service");
+            }
+
+            var productResponse = listOfDonationProducts.Select(c => new GetMyDonationDetailsProductResponse
+            {
+                Name = c.Product!.Name,
+                Quantity = c.Quantity,
+                Unit = c.Product.Unit.ToString()
+            }).ToList();
+
+            var warehouseById = await _appDbContext.Warehouses
+              .Include(c => c.Address).ThenInclude(c => c!.Country)
+              .SingleOrDefaultAsync(c => c.Id == donationById.Result!.WarehouseId);
+
+            if (warehouseById == null)
+            {
+                _logger.LogError("Couldn't find warehouse in database");
+                throw new ApplicationErrorException("Sorry there seems to be a problem with our service");
+            }
+
+            var response = new GetMyDonationDetailsResponse
+            {
+                Id = _hashids.EncodeLong(donationById.Id),
+                Name = donationById.Result!.Name,
+                CreationDate = donationById.Result.CreationDate,
+                ExpireDate = ExpireDateService.GetExpiredDate4Donation(donationById.Result.CreationDate),
```

Comment:
Ej, to miało być wyliczane przy tworzeniu dotacji, i tutaj tylko `donationById.ExpireDate`

Reply:
fixed